vm/parser.ts

```diff
@@ -711,6 +861,17 @@ module J2ME {
       }
     }
 
+    getClassNameSlow(): string {
+      return this._name || (this._name = ByteStream.readString(this.utf8Name));
+    }
```

Comment:
That's what the previous comment was about. I think old school Java like getters are the way to go. They have the advantage that they don't mutate the shape of the object and avoid the need for calling Object.defineProperty which is pretty slow itself.